pkg/pquerier/select.go

```diff
@@ -172,19 +172,44 @@ func (queryCtx *selectQueryContext) queryPartition(partition *partmgr.DBPartitio
 
 		newQuery := &partQuery{mint: mint, maxt: maxt, partition: partition, step: step}
 		if aggregationParams != nil {
-			newQuery.preAggregated = aggregationParams.CanAggregate(partition.AggrType())
-			if newQuery.preAggregated || !queryCtx.queryParams.disableClientAggr {
+			newQuery.useServerSideAggregates = aggregationParams.CanAggregate(partition.AggrType())
+			if newQuery.useServerSideAggregates || !queryCtx.queryParams.disableClientAggr {
 				newQuery.aggregationParams = aggregationParams
 			}
 		}
 
-		err = newQuery.getItems(queryCtx, metric, requestAggregatesAndRaw)
+		groupBy := queryCtx.groupBy(partition)
+		err = newQuery.getItems(queryCtx, metric, groupBy, requestAggregatesAndRaw)
 		queries = append(queries, newQuery)
 	}
 
 	return queries, err
 }
 
+func (queryCtx *selectQueryContext) groupBy(partition *partmgr.DBPartition) []string {
```

Comment:
i'd move this to `partQuery`

Reply:
But it needs the `queryParams`.

Reply:
yea well the `getItems` gets the selectQueryContext as parameter and seems like this logic should be there.. but idk